third_party/txt/src/txt/paragraph.cc

```diff
@@ -582,7 +576,12 @@ void Paragraph::Layout(double width, bool force) {
           }
 
           if (word_index < words.size() &&
-              words[word_index].end == run.start() + next_grapheme) {
+              words[word_index].start == run.start() + glyph_code_units.start) {
+            word_start_position = run_x_offset + glyph_x_offset;
+          }
```

Comment:
Thanks for moving these two blocks together while you were at it - *much* easier to follow. :)